packages/auth/src/server.ts

```diff
@@ -59,57 +59,57 @@ export const {
       },
     },
   },
-  callbacks: {
-    session: async ({ session, user }) => {
-      let userRoles: RoleTypes[] = [];
-      if (user.roles) {
-        userRoles = user.roles.reduce((acc: RoleTypes[], role) => {
-          acc.push(role.role);
-          return acc;
-        }, []);
-      }
-      if (!userRoles.includes('USER')) {
-        const updatedUser = await prisma.user.update({
-          where: {
-            id: user.id,
-          },
-          data: {
-            roles: {
-              connectOrCreate: {
-                where: {
-                  role: 'USER',
-                },
-                create: {
-                  role: 'USER',
-                },
-              },
+  adapter: {
+    ...(PrismaAdapter(prisma) as Adapter),
+    // Override createUser method to add default USER role
+    createUser: (data) =>
+      prisma.user.create({
+        data: {
+          ...data,
+          name: data.name ?? '',
+          roles: {
+            connectOrCreate: {
+              where: { role: 'USER' },
+              create: { role: 'USER' },
             },
           },
-          include: {
-            roles: true,
-          },
-        });
-        userRoles = updatedUser.roles.reduce((acc: RoleTypes[], role) => {
-          acc.push(role.role);
-          return acc;
-        }, []);
-      }
+        },
+        include: { roles: true },
+      }) as Promise<AdapterUser>,
+    // Override getSessionAndUser method to include roles. Avoids a second db query in session callback
+    getSessionAndUser: async (sessionToken) => {
+      const userAndSession = await prisma.session.findUnique({
+        where: { sessionToken },
+        include: { user: { include: { roles: true } } },
+      });
+      if (!userAndSession) return null;
+      const { user, ...session } = userAndSession;
+      return { user, session } as { session: AdapterSession; user: AdapterUser };
+    },
+  },
+  callbacks: {
+    session: async ({ session, user }) => {
       return {
         ...session,
         user: {
           ...session.user,
           id: user.id,
-          role: userRoles,
-          createAt: user.createdAt,
+          role: user.roles.map((r) => r.role),
         },
       };
     },
   },
-  adapter: PrismaAdapter(prisma) as Adapter,
   providers: [
     GitHubProvider({
       clientId: process.env.GITHUB_ID!,
       clientSecret: process.env.GITHUB_SECRET!,
+      profile: (p) =>
+        ({
+          id: p.id.toString(),
+          name: p.login,
```

Comment:
Is `.login` always the username?

We initially introduced the patch cause some users have display names like me as "Sean Boult" but most don't.

We want @<username> to always be the happy path just want to confirm this before we merge.

Reply:
related context https://github.com/typehero/typehero/issues/1210

Reply:
that’s what the patches does. uses login vs name(which is not unique)

Reply:
All the patch did was change

```diff
- name: p.name ?? p.login
+ name: p.login
```

which is what this does too